Modix.Bot/Behaviors/PromotionPollBehavior/PromotionDialogBehavior.cs

```diff
@@ -0,0 +1,211 @@
+        INotificationHandler<PromotionActionCreatedNotification>
+    {
+        private readonly TimeSpan ErrorRemoveDelay = TimeSpan.FromSeconds(5);
+
+        private readonly DateTime _utcNow = DateTime.UtcNow;
+
+        private static readonly Emoji _approveEmoji = new Emoji("👍");
+
+        private static readonly Emoji _disapproveEmoji = new Emoji("👎");
+
+        private IDesignatedChannelService _designatedChannelService { get; }
+
+        private IDiscordSocketClient _discordSocketClient { get; }
+
+        private IMemoryCache _memoryCache { get; }
+
+        private IPromotionsService _promotionsService { get; }
+
+        private IImageService _imageService { get; }
+
+        private IPromotionCampaignRepository _promotionCampaignRepository { get; }
+
+        private IAuthorizationService _authorizationService { get; }
+
+        private IPromotionDialogRepository _promotionDialogRepository { get;  }
+
+        public PromotionDialogBehavior(IDesignatedChannelService designatedChannelService,
+            IDiscordSocketClient discordSocketClient,
+            IMemoryCache memoryCache,
+            IPromotionsService promotionsService,
+            IImageService imageService,
+            IAuthorizationService authorizationService,
+            IPromotionCampaignRepository promotionCampaignRepository,
+            IPromotionDialogRepository promotionDialogRepository
+            )
+        {
+            _designatedChannelService = designatedChannelService;
+            _discordSocketClient = discordSocketClient;
+            _memoryCache = memoryCache;
+            _promotionsService = promotionsService;
+            _imageService = imageService;
+            _promotionCampaignRepository = promotionCampaignRepository;
+            _authorizationService = authorizationService;
+            _promotionDialogRepository = promotionDialogRepository;
+        }
+
+        public async Task HandleNotificationAsync(PromotionActionCreatedNotification notification,
+            CancellationToken cancellationToken = default)
+        {
+            if (cancellationToken.IsCancellationRequested)
+                return;
+
+            switch (notification.Data.Type)
+            {
+                case PromotionActionType.CampaignCreated:
+                    await CreatePromoDialog(notification.Data);
+                    break;
+                case PromotionActionType.CampaignClosed:
+                    await DeletePromoDialog(notification.Data);
+                    break;
+            }
+        }
+
+        public async Task HandleNotificationAsync(ReactionAddedNotification notification,
+            CancellationToken cancellationToken = default)
+        {
+            var reaction = notification.Reaction.Emote;
+            var user = (IGuildUser)notification.Reaction.User.Value;
+            var message = await notification.Message.GetOrDownloadAsync();
+
+            if (cancellationToken.IsCancellationRequested
```

Comment:
Generally, cancellation should be handled with `.ThrowIfCancellationRequested()`

Reply:
For these i was following the pattern in AutoRemoveMessageHandler, line 51

Reply:
Then I'd say that needs to be changed too.

In reality though, MODiX really doesn't support cancellation (not currently anyway). Wherever this token is coming from, it's likely not ever going to be triggered. I'd honestly just remove it.